benchmarks/dataset_selectivity_benchmark.py

```diff
@@ -0,0 +1,68 @@
+import copy
+import os
+
+import conbench.runner
+import pyarrow.dataset as ds
+
+from benchmarks import _benchmark
+
+
+@conbench.runner.register_benchmark
+class DatasetSelectivityBenchmark(_benchmark.Benchmark):
+    """Read and filter a dataset with different selectivity."""
+
+    name = "dataset-selectivity"
+    sources = ["nyctaxi_multi_parquet_s3", "chi_traffic_2020_Q1"]
+    sources_test = ["nyctaxi_multi_parquet_s3_sample", "chi_traffic_sample"]
+    valid_cases = (["selectivity"], ["1%"], ["10%"], ["100%"])
+    filters = {
+        "nyctaxi_multi_parquet_s3": {
+            "1%": ds.field("pickup_longitude") < -74.013451,  # 561384
+            "10%": ds.field("pickup_longitude") < -74.002055,  # 5615432
+            "100%": None,  # 56154689
+        },
+        "chi_traffic_2020_Q1": {
+            "1%": ds.field("END_LONGITUDE") < -87.807262,  # 124530
+            "10%": ds.field("END_LONGITUDE") < -87.7624,  # 1307565
+            "100%": None,  # 13038291
+        },
+        "nyctaxi_multi_parquet_s3_sample": {
+            "1%": ds.field("pickup_longitude") < -74.0124,  # 20
+            "10%": ds.field("pickup_longitude") < -74.00172,  # 200
+            "100%": None,  # 2000
+        },
+        "chi_traffic_sample": {
+            "1%": ds.field("END_LONGITUDE") < -87.80726,  # 10
+            "10%": ds.field("END_LONGITUDE") < -87.76148,  # 100
+            "100%": None,  # 1000
+        },
+    }
+    options = copy.deepcopy(_benchmark.Benchmark.options)
+    options["cpu_count"]["default"] = len(os.sched_getaffinity(0))
```

Comment:
If `cpu_count` isn't specified, the max will be used (`None` == max available).

So I think the following lines can be omitted.

```
    options = copy.deepcopy(_benchmark.Benchmark.options)
    options["cpu_count"]["default"] = len(os.sched_getaffinity(0))
```

Same for the f-string in the test and the other call to `sched_getaffinity`.

PS. This might be the best 1st time contributor pull request I've ever seen. You followed all the existing patterns, improved the code without enforcing a new style, and advanced the project with a reviewable sized pull request. Thanks!

I would love to know more about you, how you found our small repo, your motivation for contributing, and any pain points you encountered trying to submit this pull request (so we can make it better). There's lots I still want to improve about the design, and I'm really impressed that you figured it all out despite some of my WIP design choices.

PS. If you don't want to answer publicly, my email is: diana.joan.clarke@gmail.com

Reply:
Thanks for letting me know. I checked the arrow code before adding the configuration for the CPU count, but obviously, I didn't read it carefully enough. 

I'm a Ph.D. student working on a project about filtering data in parquet files using Arrow. One thing I'm currently interested in is how much performance difference among different systems (e.g. x86 vs. embedded systems) with respect to the selectivity ratio. I found that Conbench is written to benchmark Arrow, so I think why not write the test using this framework.

Reply:
If it's not too much burden I'm sure we'd love for you to share your project results (when you're finished) on one of the Arrow mailing lists :)

Reply:
Sure, I will post my results by the end of this week.

Reply:
I asked my coworker from national labs about the results. He suggested not to show up the results at this moment because they may be useful for a later publication. So far we got instructive results from benchmarking on a SmartNIC. I'll happy to refer to this benchmark in the publication and let you know once we get to that step.

Reply:
No problem.  Best of luck and let us know if we can be of any help.